tracer/src/Datadog.Trace/DuckTyping/DuckType.Statics.cs

```diff
@@ -185,6 +187,74 @@ static ModuleBuilder CreateModuleBuilder(string name, Assembly targetAssembly)
             }
         }
 
+        private static Type? GetTypeFromPartialName(string partialName, bool throwOnError = false)
+        {
+            Type? type = null;
+            ExceptionDispatchInfo? loadException = null;
+
+            try
+            {
+                // Let's try to find the type using the partial name first.
+                type = Type.GetType(partialName, throwOnError: throwOnError);
+            }
+            catch (Exception ex)
+            {
+                // let's capture the exception
+                loadException = ExceptionDispatchInfo.Capture(ex);
+            }
```

Comment:
This feels kinda weird. I know it's because you're trying to keep existing behaviour, but do we _need_ to? 🤔 can't we just get it with `throwOnError:false` (which means we can extract the default behaviour more easily, as above), and then throw later if necessary...